File: Practice/Anagrams/Anagrams.py
import sys
if len(sys.argv) > 0 :
	anagrams = {}
	with open(sys.argv[1], 'r') as file:
		list(map(lambda x: anagrams.setdefault(''.join((sorted(x.strip().lower()))), set()).add(x.strip()), file))

	list(map(lambda x : print(x), anagrams.values()))

File: Practice/Anagrams/Anagrams.rb
if !ARGV.empty?
	anagrams = Hash.new{|h, k| h[k] = []}
	IO.foreach(ARGV[0]){|w| anagrams[w.strip.downcase.chars.sort.join].push(w.strip)}
	anagrams.values.each{|v| print v, "\n"}
end

